src/GlobalPayments.Api/Terminals/INGENICO/IngenicoController.cs

```diff
@@ -0,0 +1,288 @@
+            if (_device == null) {
+                _device = new IngenicoInterface(this);
+            }
+            return _device;
+        }
+        internal override IDeviceCommInterface ConfigureConnector() {
+            switch (_settings.ConnectionMode) {
+                case ConnectionModes.SERIAL:
+                    return new IngenicoSerialInterface(_settings);
+                case ConnectionModes.TCP_IP:
+                case ConnectionModes.SSL_TCP:
+                case ConnectionModes.HTTP:
+                case ConnectionModes.TCP_IP_SERVER:
+                    return new IngenicoTcpInterface(_settings);
+                default:
+                    throw new NotImplementedException();
+            }
+        }
+
+        #region overrides
+        internal override ITerminalResponse ManageTransaction(TerminalManageBuilder builder) {
+            var request = BuildManageTransaction(builder);
+
+            if (builder.TransactionType == TransactionType.Cancel) {
+                return DoCancelRequest(request);
+            }
+            else if (builder.TransactionType == TransactionType.Reversal) {
+                return DoReverseRequest(request);
+            }
+            else {
+                return DoRequest(request);
+            }
+        }
+
+        internal override ITerminalReport ProcessReport(TerminalReportBuilder builder) {
+            var request = TerminalUtilities.BuildIngenicoRequest(INGENICO_REQ_CMD.RECEIPT.FormatWith(builder.ReceiptType), _settings.ConnectionMode);
+            return ReportRequest(request);
+        }
+
+        internal override ITerminalResponse ProcessTransaction(TerminalAuthBuilder builder) {
+            var request = BuildRequestMessage(builder);
+            return DoRequest(request);
+        }
+
+        internal override byte[] SerializeRequest(TerminalAuthBuilder builder) {
+            return BuildRequestMessage(builder).GetSendBuffer();
+        }
+
+        internal override byte[] SerializeRequest(TerminalManageBuilder builder) {
+            throw new NotImplementedException();
+        }
+
+        internal override byte[] SerializeRequest(TerminalReportBuilder builder) {
+            throw new NotImplementedException();
+        }
+        #endregion
+
+        #region Methods
+        internal IDeviceMessage BuildManageTransaction(TerminalManageBuilder builder) {
+            var _refNumber = builder.ReferenceNumber;
+            var _amount = ValidateAmount(builder.Amount);
+            var _returnRep = 1;
+            var _paymentMode = ValidatePaymentMode(builder.PaymentMode);
+            var _paymentType = (int?)((IngenicoInterface)_device).paymentMethod ?? 0;
+            var _currencyCode = (string.IsNullOrEmpty(builder.CurrencyCode) ? "826" : builder.CurrencyCode);
+            var _privData = "EXT0100000";
+            var _immediateAns = 0;
+            var _forceOnline = 0;
+            var _extendedData = "0000000000";
```

Comment:
var should not be used when the datatype is unknown or ambiguous. e.g. var _paymentMode = ValidatePaymentMode(...); 

is it unclear from the above line of code what datatype is being stored in _paymentMode without having to go look at the ValidatePaymentMode method.